### PR TITLE
Working on the ssl benchmark docs

### PR DESCRIPTION
**Pull request by calebrob6**
### Summary

Adds a new doc page describing how to benchmark SSL tasks. See page here: https://torchgeo--4001.org.readthedocs.build/en/4001/user/ssl_benchmark.html

See the tutorial notebook rendering here: https://github.com/torchgeo/torchgeo/blob/ssl-benchmark-docs/docs/tutorials/ssl_knn_eval.ipynb

### Rationale

SSL trainers can't be compared by their training loss, and knowing if an SSL trainer is "correct" requires using it on real data not just looking at its code. E.g. using SimCLR with EuroSAT with its default learning rate reached an embedding standard deviation of 0.0000 and a mean pairwise cosine similarity of 1.0000 (i.e. the representation collapsed completely), while its loss fell 30% from initial. Previously, there was no documented or standardized way to check, so this page defines one.

### Checklist

- [x] I have read the [Contributing docs](https://docs.torchgeo.org/en/stable/user/contributing.html)
- [x] I have read the [AI policy](https://github.com/torchgeo/governance/blob/main/AI-POLICY.md)

- [ ] 🟢 **No AI usage**: written by humans, for humans
- [x] 🟡 **AI-assisted**: AI helped with the coding, but I understand every line
- [ ] 🔴 **AI-generated**: AI did everything, review with caution

### CODE REVIEW

File: docs/tutorials/basic_usage.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ The following tutorials introduce the basic concepts and components of TorchGeo:
 * `Spectral Indices <indices.ipynb>`_: Visualizing and appending spectral indices
 * `Pretrained Weights <pretrained_weights.ipynb>`_: Models and pretrained weights
 * `Lightning Tasks <tasks.ipynb>`_: PyTorch Lightning data modules and tasks
+* `Evaluating SSL Checkpoints <ssl_knn_eval.ipynb>`_: Scoring a self-supervised encoder with a kNN probe
```
Comment (adamjstewart) [inline docs/tutorials/basic_usage.rst:10]: Wondering if this is more of a Case Study, doesn't feel so basic

File: docs/user/ssl_benchmark.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
+Self-Supervised Learning Tasks
```
Comment (adamjstewart) [inline docs/user/ssl_benchmark.rst:1]: Some of this stuff makes more sense in `docs/api/tasks.rst`, some of it makes more sense in `docs/user/contributing.rst`, some of it makes more sense in a Customization tutorial for adding a new SSL task.

File: docs/user/ssl_benchmark.rst [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+Adding a new task
+-----------------
+
+When adding an SSL task, include:
+
+#. ``torchgeo/tasks/foo.py``, subclassing :class:`~torchgeo.tasks.BaseTask`.
+#. An entry in ``torchgeo/tasks/__init__.py``.
+#. Tests in ``tests/tasks/test_foo.py``.
+#. An API documentation entry in ``docs/api/tasks.rst``.
```
Comment (adamjstewart) [inline docs/user/ssl_benchmark.rst:163]: Should we have guidelines for which datamodules to use for testing these tasks?

File: tests/configs/ssl_benchmarking/byol_resnet50.yaml
Comment (adamjstewart) [inline tests/configs/ssl_benchmarking/byol_resnet50.yaml:1]: Not really sure if I want a new `tests/configs/ssl_benchmarking` directory, why not just `tests/conf` like everyone else?

**Review by adamjstewart — COMMENTED** [review]
Didn't look at this one very closely, just some high-level comments on where things belong in the docs. Hoping someone else can review the text more closely and actually run the tutorial locally.
